### PR TITLE
feat: user authz

### PR DESCRIPTION
**Pull request by oh2024**
* **What kind of change does this PR introduce?** 
feature

* **What is the current behavior?** (You can also link to an open issue here)
Grant/Revoke statements are not supported at all

* **What is the new behavior (if this is a feature change)?**
* Grant/Revoke is supported for the CREATE USER and DROP USER statements
* Grant options is supported and is used to give permissions to grant

### CODE REVIEW

File: hybridse/include/node/node_enum.h [hunk 1]
```diff
@@ -98,6 +98,8 @@ enum SqlNodeType {
     kColumnSchema,
     kCreateUserStmt,
     kAlterUserStmt,
+    kGrantStmt,
+    kRevokeStmt,
```
Comment (aceforeverd) [inline hybridse/include/node/node_enum.h:102]: append new enums only

File: hybridse/src/planv2/ast_node_converter.cc [hunk 2]
```diff
@@ -2133,6 +2148,81 @@ base::Status ConvertAlterUserStatement(const zetasql::ASTAlterUserStatement* roo
...
+    for (auto grantee : root->grantee_list()->grantee_list()) {
+        if (grantee == nullptr) {
+            continue;
+        }
+
+        std::string grantee_str;
+        CHECK_STATUS(AstStringLiteralToString(grantee, &grantee_str));
+        grantees.push_back(grantee_str);
+    }
+    *output = node_manager->MakeNode<node::GrantNode>(target_type, target_path.at(0), target_path.at(1), privileges,
```
Comment (aceforeverd) [inline hybridse/src/planv2/ast_node_converter.cc:2183]: what if `target_path` size = 1 ?

File: src/nameserver/system_table.h [hunk 3]
```diff
@@ -163,20 +163,16 @@ class SystemTable {
                 break;
             }
             case SystemTableType::kUser: {
-                SetColumnDesc("host", type::DataType::kString, table_info->add_column_desc());
-                SetColumnDesc("user", type::DataType::kString, table_info->add_column_desc());
-                SetColumnDesc("password", type::DataType::kString, table_info->add_column_desc());
+                SetColumnDesc("Host", type::DataType::kString, table_info->add_column_desc());
```
Comment (aceforeverd) [inline src/nameserver/system_table.h:166]: Is there any usage over this table from the previous release ? If so it would a breaking change

Comment (oh2024) [reply src/nameserver/system_table.h:166]: Yes it's a breaking change

**Review by vagetablechicken — APPROVED** [review]

File: hybridse/src/planv2/ast_node_converter.cc (hunk 2 above)
Comment (oh2024) [reply hybridse/src/planv2/ast_node_converter.cc:2183]: in grant statements the target_path is required to have 2 elements: database and target

**Review by tobegit3hub — APPROVED** [review]
LGTM

**Pull request merged**